### DISCUSSION TITLE
What should I use for derived state selectors?

### DISCUSSION THREAD

**Discussion by ronny-lark** [pmndrs/zustand#2108] [Q&A]
Say I have a case like this:
```ts
const bearId = 'id';
const selectedBear = useBearStore(state => state.bear.find(bear => bear.id === bearId));
```
Will this trigger unnecessary rerenders like when using objects/arrays?
In those cases it is suggested to use `useShallow`, but what about this case, where only one state is being used?
What about a similar case where the `bearId` is a state:
```ts
useBearStore(state => state.bears.find(bear => bear.id === state.selectedBearId));
```
Do I need `useShallow` in any of these 2 cases?

**Comment by dbritto-dev** [comment]
@ronny-lark yes, you'd better using `useShallow`
